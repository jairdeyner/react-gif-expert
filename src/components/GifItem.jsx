import PropTypes from "prop-types";

export const GifItem = ({ gif }) => {
  return (
    <div className="card">
      <img src={gif.url} alt={gif.title} />
      <p>{gif.title}</p>
    </div>
  );
};

GifItem.propTypes = {
  gif: PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  }).isRequired,
};
