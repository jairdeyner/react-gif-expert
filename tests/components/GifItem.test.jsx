import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Test en <GifItem />", () => {
  const gif = {
    id: "2",
    title: "Naruto",
    url: "https://naruto.png/",
  };

  test("debe hacer match snapshot", () => {
    const { container } = render(<GifItem gif={gif} />);

    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con el URL y el ALT inidcado", () => {
    render(<GifItem gif={gif} />);
    // screen.debug();

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(gif.url);
    expect(alt).toBe(gif.title);
  });

  test("debe de mostrar el tÃ­tulo en el componente", () => {
    render(<GifItem gif={gif} />);

    expect(screen.getByText(gif.title)).toBeTruthy();
  });
});
