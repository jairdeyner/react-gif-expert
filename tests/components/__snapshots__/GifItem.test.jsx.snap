// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GifItem /> debe hacer match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://naruto.png/"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
